addons/rating/models/rating.py

```diff
@@ -26,7 +26,8 @@ def _selection_target_model(self):
     res_name = fields.Char(string='Resource name', compute='_compute_res_name', store=True, help="The name of the rated resource.")
     res_model_id = fields.Many2one('ir.model', 'Related Document Model', index=True, ondelete='cascade', help='Model of the followed resource')
     res_model = fields.Char(string='Document Model', related='res_model_id.model', store=True, index=True, readonly=True)
-    res_id = fields.Integer(string='Document', required=True, help="Identifier of the rated object", index=True)
+    res_id = fields.Many2oneReference(string='Document', model_field='res_model', help="Identifier of the rated object",
```

Comment:
@tde-banana-odoo Following discussions with @awa-odoo @ryv-odoo and @rco-odoo  , I converted the integer res_id field to a Many2oneReference. The cache invalidation in the test is not necessary anymore. The order in the cache can't be ensured but the new compute method force the ordering in the SQL request. The final result is more clear. Thanks to all the person who answered my questions to understand this issue :-).